src/iadmin.cpp

```diff
@@ -1624,9 +1625,13 @@ doCommand( char *cmdToken[], rodsArguments_t* _rodsArgs = 0 ) {
         return 0;
     }
     if ( strcmp( cmdToken[0], "rmgroup" ) == 0 ) {
-        generalAdmin( 0, "rm", "user", cmdToken[1],
-                      myEnv.rodsZone, "", "", "", "", "", "" );
-        return 0;
+      const auto version = irods::to_version(Conn->svrVersion->relVersion);
+      const char *entity_type =
+          (version && version.value() > irods::version{4, 3, 1}) ? "group"
+                                                                 : "user";
+      generalAdmin(0, "rm", const_cast<char*>(entity_type), cmdToken[1], myEnv.rodsZone, "", "",
```

Comment:
The need for `const_cast` sucks, but I get it. I can never remember if casting away const is UB. I bet it is.

To get around it, allocate a buffer on the stack, copy the string into it, and pass the buffer to generalAdmin.

Alternatively, assign the string to a non-const std::string and use `.data()` to get at the underlying buffer.

Reply:
Hmm, a bit of research turns up that it's only UB if you try to modify it through the un-const'ed pointer, I think? Would it be legal in this case, then, since I don't suppose `generalAdmin` is going to modify its 3rd argument anyhow (since string literals were passed in its place before...)
I had originally wanted to do this because I wanted to avoid the extra copy/alloc if it wasn't necessary.
Or do you think it's just not worth the trouble and I should just do the copy/alloc?

Reply:
That's fair. I've used `const_cast` in code under the assumption that nothing will modify it.

Letting it ride.